src/smc-webapp/custom-software/selector.tsx

```diff
@@ -0,0 +1,273 @@
+import { Component, React, Rendered } from "../app-framework";
+import { ComputeImages, ComputeImage, ComputeImageTypes } from "./init";
+const { SiteName, CompanyName, HelpEmailLink } = require("../customize");
+const { Markdown, SearchInput, Icon } = require("../r_misc");
+import { CUSTOM_SOFTWARE_HELP_URL } from "./util";
+
+const BINDER_URL = "https://mybinder.readthedocs.io/en/latest/";
+
+const legacy: ComputeImageTypes = "legacy";
+const custom: ComputeImageTypes = "custom";
+
+const COLORS = require("smc-util/theme").COLORS;
+
+const {
+  Row,
+  Col,
+  FormGroup,
+  ControlLabel,
+  ListGroup,
+  ListGroupItem,
+  Radio
+} = require("react-bootstrap");
+
+interface CSProps {
+  images?: ComputeImages;
+  // this should be Partial<State> ?
+  setParentState: (obj: any) => void;
+  image_type: ComputeImageTypes;
+  image_selected?: string;
+  // toggles form true → false after first edit
+  title_prefill: boolean;
+}
+
+interface CSState {
+  search_img: string;
+}
+
+const CS_INIT_STATE: Readonly<CSState> = Object.freeze({
+  search_img: ""
+});
+
+const cs_list_style: Readonly<React.CSSProperties> = Object.freeze({
```

Comment:
Freezing CSS properties seems kind of silly to me. What's the worry here? Or do you just wish (as I do) that javascript had better immutable objects?